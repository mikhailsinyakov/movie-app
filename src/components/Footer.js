import React from "react";
import { withLanguageContext } from "./Language";

const Footer = ({ getUIText }) => (
  <footer>
    <a href="https://github.com/mikhailsinyakov">Github</a>
    <span>{getUIText("tmdbAttribution")}</span>
  </footer>
);

export default withLanguageContext(Footer);
