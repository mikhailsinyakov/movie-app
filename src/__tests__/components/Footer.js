import React from "react";
import { create } from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import withLanguage from "../../components/Language";
import Footer from "../../components/Footer";

const Parent = ({ children }) => <div>{children}</div>;
const ParentWithLanguage = withLanguage(Parent);

let component, instance;
beforeAll(() => {
  component = create(
    <Router>
      <ParentWithLanguage>
        <Footer />
      </ParentWithLanguage>
    </Router>
  );
  instance = component.root;
});

it("matches snapshot", () => {
  expect(component.toJSON()).toMatchSnapshot();
});

it("has link and text", () => {
  const link = instance.findByType("a");
  const text = instance.findByType("span");
  expect(link).toBeTruthy();
  expect(text).toBeTruthy();
});
