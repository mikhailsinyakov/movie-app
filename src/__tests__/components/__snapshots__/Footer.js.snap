// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <footer>
    <a
      href="https://github.com/mikhailsinyakov"
    >
      Github
    </a>
    <span>
      This product uses the TMDb API but is not endorsed or certified by TMDb
    </span>
  </footer>
</div>
`;
